### PR TITLE
[azure] Add retries for Azure client

### PR DESCRIPTION
**Pull request by GiriB**
Azure client today doesn't implement any retries. When working with bigger repos, we see (transient) ServiceUnavailable often. This PR adds retries for the standard 500, 502 and 503 errors.

### CODE REVIEW

File: common/lib/dependabot/clients/azure.rb [hunk 1]
```diff
@@ -8,6 +8,14 @@ module Clients
     class Azure
       class NotFound < StandardError; end
 
+      class InternalServerError < StandardError; end
```
Comment (GiriB) [inline common/lib/dependabot/clients/azure.rb:11]: Defined these to throw better error messages.

Comment (GiriB) [inline common/lib/dependabot/clients/azure.rb:11]: Added these to make up better stacktrace. Currently errors are show like `JSON::ParseError` while parsing failed response body.

**Comment by GiriB** [comment]
@jurre @brrygrdn Can you help review this? Thanks!

File: common/lib/dependabot/clients/azure.rb [hunk 2]
```diff
@@ -27,10 +35,11 @@ def self.for_source(source:, credentials:)
       # Client #
       ##########
 
-      def initialize(source, credentials)
+      def initialize(source, credentials, max_retries: 3)
         @source = source
         @credentials = credentials
         @auth_header = auth_header_for(credentials&.fetch("token", nil))
+        @max_retries = max_retries || 3
```
Comment (jurre) [inline common/lib/dependabot/clients/azure.rb:42]: This is redundant as `max_retries` already has a default argument of 3, but I noticed the other retryable clients follow the same pattern so feel free to leave it in for now

**Review by jurre — APPROVED** [review]
Looks great 👍 I'm going to go ahead and merge this, but probably won't cut a new release until after the holidays, as we will be very short staffed and have a short release-freeze.

File: common/lib/dependabot/clients/azure.rb (hunk 2 above)
Comment (GiriB) [reply common/lib/dependabot/clients/azure.rb:42]: Thanks!

**Pull request merged**